cmd/faucet/faucet.go

```diff
@@ -272,6 +282,21 @@ func (f *faucet) webHandler(w http.ResponseWriter, r *http.Request) {
 
 // apiHandler handles requests for Ether grants and transaction statuses.
 func (f *faucet) apiHandler(w http.ResponseWriter, r *http.Request) {
+	ip := r.RemoteAddr
+	if len(r.Header.Get("X-Forwarded-For")) > 0 {
+		ips := strings.Split(r.Header.Get("X-Forwarded-For"), ",")
+		if len(ips) > 0 {
+			ip = strings.TrimSpace(ips[len(ips)-1])
```

Comment:
why only limit the last ip?

Reply:
there is a potential risk that `X-Forwarded-For` can be faked.

Reply:
Usually it corresponds to the original ip address without port. 

Yes `X-Forwarded-For` can be faked but it adds an extra step for an attacker. One extra thing I can think of is to do a validation check on the `ips` unless you have any other relevant suggestion?